### PR TITLE
Create API for map pins search

### LINKED ISSUE #3916: [feature request] Create API for map pins search
**Is your feature request related to a problem? Please describe.**
Move our current map-pins search capability to an API using Remix routing.
Create a Remix loader in a api.map.ts route.

**Additional context**
This is part of the effort to decouple the firebase search from the client.
And make it easier for a Supabase migration and improve codebase maintainability.

### PR DESCRIPTION
**Pull request by goratt12**
## PR Checklist

- [ ] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)
- [ ] - Unit and e2e tests for the changes that have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix (fixes an issue)
- [ ] Feature (adds functionality)
- [ ] Code style update
- [X] Refactoring (no functional changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation changes
- [ ] Other... Please describe:

## What is the current behavior?
map pins fetched from the old API (platfrom-api)

## What is the new behavior?

map pins is fetching using Remix backend API

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Git Issues

Closes #3916 

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of our monthly maintainers call (first Monday of the month)

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

File: src/routes/api.mappins.$id.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    const formattedMapPin = formatMapPinForResponse(mapPin)
+    const userProfile = await getUserProfile(id)
+    const userData = userProfile || {}
+
+    let heroImageUrl = ''
+    if (userData.coverImages && userData.coverImages.length) {
+      heroImageUrl = userData.coverImages[0].downloadUrl
+    }
+
+    // TODO: Load avatar from user profile
```
Comment (mariojsnunes) [inline src/routes/api.mappins.$id.ts:28]: TODO, is this for another PR?

File: src/routes/api.mappins.$id.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+async function getUserProfile(userId) {
+  const collectionRef1 = collection(firestore, DB_ENDPOINTS.users)
+  const usersQuery1 = query(collectionRef1, where('_authId', '==', userId))
+  const queryResults1 = await getDocs(usersQuery1)
+
+  if (queryResults1.docs.length === 1) {
+    return queryResults1.docs[0].data()
+  }
+
+  const collectionRef2 = collection(firestore, DB_ENDPOINTS.users)
```
Comment (mariojsnunes) [inline src/routes/api.mappins.$id.ts:80]: if querying for _authId is really needed, the collection is the same, so we don't need a "collectionRef2"

File: src/routes/api.mappins.$id.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  const usersQuery2 = query(collectionRef2, where('_id', '==', userId))
+  const queryResults2 = await getDocs(usersQuery2)
```
Comment (mariojsnunes) [inline src/routes/api.mappins.$id.ts:82]: if querying for _authId is really needed:

```
const usersCollection = collection(firestore, DB_ENDPOINTS.users);
const usersByAuthId = await getDocs(query(usersCollection, where('_authId', '==', userId)));
const usersById = await getDocs(query(usersCollection, where('_id', '==', userId)));
```

File: src/routes/api.mappins.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+const cache = new Keyv<IMapPin[]>({ ttl: 600000 }) // ttl: 10 minutes
+
+// runs on the server
+export const loader = async () => {
+  const cachedMappins = await cache.get('mappins')
+
+  // check if cached map pins are availbe, if not - load from db and cache them
+  if (cachedMappins) {
+    return json({ mapPins: cachedMappins })
+  } else {
```
[the remark this answers is not in the document]
Comment (goratt12) [reply src/routes/api.mappins.ts]: Fixed :)

File: src/routes/api.mappins.$id.ts (hunk 3 above)
Comment (goratt12) [reply src/routes/api.mappins.$id.ts]: Fixed :)

File: src/routes/api.mappins.$id.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    _deleted: !!data._deleted,
+    type: data.type || data.pinType,
+    verified: !!data.verified,
+  }
+}
+
+// taken from platform-api
+async function getUserProfile(userId) {
+  const collectionRef1 = collection(firestore, DB_ENDPOINTS.users)
+  const usersQuery1 = query(collectionRef1, where('_authId', '==', userId))
```
[the remark this answers is not in the document]
Comment (benfurber) [reply src/routes/api.mappins.$id.ts]: I believe _authId relates to the auth stuff which is linked to a user recond on sign-up. Not sure why it's needed here...

**Review by mariojsnunes — APPROVED** [review]

**Pull request merged**